chia/full_node/mempool_check_conditions.py

```diff
@@ -132,40 +134,222 @@ def mempool_assert_my_amount(condition: ConditionWithArgs, unspent: CoinRecord)
+    if condition is ConditionOpcode.AGG_SIG_UNSAFE or condition is ConditionOpcode.AGG_SIG_ME:
+        return ConditionCost.AGG_SIG.value, parse_aggsig(args)
+    elif condition is ConditionOpcode.CREATE_COIN:
+        return ConditionCost.CREATE_COIN.value, parse_create_coin(args)
+    elif condition is ConditionOpcode.ASSERT_SECONDS_ABSOLUTE:
+        return ConditionCost.ASSERT_SECONDS_ABSOLUTE.value, parse_seconds(args, Err.ASSERT_SECONDS_ABSOLUTE_FAILED)
+    elif condition is ConditionOpcode.ASSERT_SECONDS_RELATIVE:
+        return ConditionCost.ASSERT_SECONDS_RELATIVE.value, parse_seconds(args, Err.ASSERT_SECONDS_RELATIVE_FAILED)
+    elif condition is ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE:
+        return ConditionCost.ASSERT_HEIGHT_ABSOLUTE.value, parse_height(args, Err.ASSERT_HEIGHT_ABSOLUTE_FAILED)
+    elif condition is ConditionOpcode.ASSERT_HEIGHT_RELATIVE:
+        return ConditionCost.ASSERT_HEIGHT_RELATIVE.value, parse_height(args, Err.ASSERT_HEIGHT_RELATIVE_FAILED)
+    elif condition is ConditionOpcode.ASSERT_MY_COIN_ID:
+        return ConditionCost.ASSERT_MY_COIN_ID.value, parse_coin_id(args, Err.ASSERT_MY_COIN_ID_FAILED)
+    elif condition is ConditionOpcode.RESERVE_FEE:
+        return ConditionCost.RESERVE_FEE.value, parse_fee(args)
+    elif condition is ConditionOpcode.CREATE_COIN_ANNOUNCEMENT:
+        return ConditionCost.CREATE_COIN_ANNOUNCEMENT.value, parse_announcement(args)
+    elif condition is ConditionOpcode.ASSERT_COIN_ANNOUNCEMENT:
+        return ConditionCost.ASSERT_COIN_ANNOUNCEMENT.value, parse_hash(args, Err.ASSERT_ANNOUNCE_CONSUMED_FAILED)
+    elif condition is ConditionOpcode.CREATE_PUZZLE_ANNOUNCEMENT:
+        return ConditionCost.CREATE_PUZZLE_ANNOUNCEMENT.value, parse_announcement(args)
+    elif condition is ConditionOpcode.ASSERT_PUZZLE_ANNOUNCEMENT:
+        return ConditionCost.ASSERT_PUZZLE_ANNOUNCEMENT.value, parse_hash(args, Err.ASSERT_ANNOUNCE_CONSUMED_FAILED)
+    elif condition is ConditionOpcode.ASSERT_MY_PARENT_ID:
+        return ConditionCost.ASSERT_MY_PARENT_ID.value, parse_coin_id(args, Err.ASSERT_MY_PARENT_ID_FAILED)
+    elif condition is ConditionOpcode.ASSERT_MY_PUZZLEHASH:
+        return ConditionCost.ASSERT_MY_PUZZLEHASH.value, parse_hash(args, Err.ASSERT_MY_PUZZLEHASH_FAILED)
+    elif condition is ConditionOpcode.ASSERT_MY_AMOUNT:
+        return ConditionCost.ASSERT_MY_AMOUNT.value, parse_amount(args)
+    else:
+        raise ValidationError(Err.INVALID_CONDITION)
+
+
+opcodes: Set[bytes] = set(item.value for item in ConditionOpcode)
+
+
+def parse_condition(cond: SExp, safe_mode: bool) -> Tuple[int, Optional[ConditionWithArgs]]:
+    condition = cond.first().as_atom()
+    if condition in opcodes:
+        opcode: ConditionOpcode = ConditionOpcode(condition)
+        cost, args = parse_condition_args(cond.rest(), opcode)
+        cvl = ConditionWithArgs(opcode, args) if args is not None else None
+    elif not safe_mode:
+        opcode = ConditionOpcode.UNKNOWN
```

Comment:
I don't know if this is new functionality, but I don't like the idea of casting unknown opcodes to a fixed "unknown" value... it's just asking for trouble later.

Reply:
this is not new functionality: https://github.com/Chia-Network/chia-blockchain/pull/6654/files#diff-c83e8da0205224011f82095884364d75a548ebc60f5beba113b286a62c42921bL157

In a future patch I'm simply throwing away unknown conditions. We don't do anything with them anywhere anyway (and we can't, because we don't know what it is)